### PR TITLE
Feature/st 39 session timeout

### PR DESCRIPTION
**Pull request by raxitp02**
Session Timeout for 15 min

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,13 +31,17 @@
     "@ionic/angular": "^5.1.0",
     "@ionic/pwa-elements": "^1.5.2",
     "@logisticinfotech/ionic4-datepicker": "^1.4.3",
+    "@ng-idle/core": "^8.0.0-beta.4",
+    "@ng-idle/keepalive": "^8.0.0-beta.4",
```
Comment (Dhruvkumarshah) [inline package.json:35]: Can we check if there is stable version for idle? If yes, please update dependencies to stable version.

File: src/app/app.module.ts [hunk 2]
```diff
@@ -40,7 +50,15 @@ import { Ionic4DatepickerModule } from '@logisticinfotech/ionic4-datepicker';
...
-    Ionic4DatepickerModule
+    Ionic4DatepickerModule,
+    NgIdleKeepaliveModule.forRoot(),
+    MomentModule,
+
+
+    BrowserAnimationsModule,
+    NgIdleKeepaliveModule.forRoot(),
+    HttpClientModule,
+    ModalModule.forRoot(),
```
Comment (Dhruvkumarshah) [inline src/app/app.module.ts]: What is the use of ModalModule here?

File: src/app/services/auth.service.ts [hunk 3]
```diff
@@ -1,20 +1,45 @@
...
+  idleState = 'Not started.';
+  timedOut = false;
+  lastPing?: Date = null;
 
-  constructor(private afAuth: AngularFireAuth) {}
+  constructor(
+    private afAuth: AngularFireAuth,
+    private idle: Idle,
+    private ngZone: NgZone,
+    public alertController: AlertController,
```
Comment (Dhruvkumarshah) [inline src/app/services/auth.service.ts]: Make it private

File: package.json (hunk 1 above)
Comment (raxitp02) [reply package.json:35]: https://www.npmjs.com/package/@ng-idle/core
https://www.npmjs.com/package/@ng-idle/keepalive

All versions are stable. They use that kind of name for all version. Check those links.

File: src/app/services/auth.service.ts (hunk 3 above)
Comment (raxitp02) [reply src/app/services/auth.service.ts]: changed to private

**Review by Dhruvkumarshah — APPROVED** [review]

**Review by Sharmil7171 — APPROVED** [review]

**Pull request merged**